Source/Plugins/Core/com.equella.core/js/__stories__/search/SearchResultList.stories.tsx

```diff
@@ -0,0 +1,99 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import type { Meta, Story } from "@storybook/react";
+import * as React from "react";
+import {
+  getEmptySearchResult as emptySearch,
+  getSearchResult as singlePageSearch,
+} from "../../__mocks__/getSearchResult";
+import { defaultSearchOptions } from "../../tsrc/modules/SearchModule";
+import {
+  SearchResultList,
+  SearchResultListProps,
+} from "../../tsrc/search/components/SearchResultList";
+import { action } from "@storybook/addon-actions";
+
+export default {
+  title: "Search/SearchResultList",
+  component: SearchResultList,
+  argTypes: {
+    onClearSearchOptions: { action: "onClearSearchOptions called" },
+  },
+} as Meta<SearchResultListProps>;
+
+const sharedPaginationArgs = {
+  currentPage: defaultSearchOptions.currentPage,
+  rowsPerPage: defaultSearchOptions.rowsPerPage,
+  /* I wasn't able to get nested actions to work inside argTypes, so falling back to the old addon-actions for these. See https://github.com/storybookjs/storybook/issues/11525 and https://github.com/storybookjs/storybook/issues/10979#issuecomment-657640744*/
+  onPageChange: action("onRowsPerPageChange"),
+  onRowsPerPageChange: action("onRowsPerPageChange"),
+};
+
+const sharedOrderSelectProps = {
+  value: defaultSearchOptions.sortOrder,
+  /* I wasn't able to get nested actions to work inside argTypes, so falling back to the old addon-actions for these. See https://github.com/storybookjs/storybook/issues/11525 and https://github.com/storybookjs/storybook/issues/10979#issuecomment-657640744*/
+  onChange: action("onChange called"),
+};
+
+export const EmptyResultListComponent: Story<SearchResultListProps> = (
+  args
+) => <SearchResultList {...args}></SearchResultList>;
+
+EmptyResultListComponent.args = {
+  searchResultItems: emptySearch.results,
+  showSpinner: false,
+  paginationProps: {
+    ...sharedPaginationArgs,
+    count: emptySearch.available,
+  },
+  orderSelectProps: {
+    ...sharedOrderSelectProps,
+  },
+};
+
+export const BasicSearchResultListComponent: Story<SearchResultListProps> = (
+  args
+) => <SearchResultList {...args}></SearchResultList>;
+
+BasicSearchResultListComponent.args = {
+  searchResultItems: singlePageSearch.results,
+  showSpinner: false,
+  paginationProps: {
+    ...sharedPaginationArgs,
+    count: singlePageSearch.available,
+  },
+  orderSelectProps: {
+    ...sharedOrderSelectProps,
+  },
+};
+
+export const LoadingSearchResultListComponent: Story<SearchResultListProps> = (
+  args
+) => <SearchResultList {...args}></SearchResultList>;
+
+LoadingSearchResultListComponent.args = {
+  searchResultItems: singlePageSearch.results,
+  showSpinner: true,
+  paginationProps: {
+    ...sharedPaginationArgs,
+    count: singlePageSearch.available,
+  },
+  orderSelectProps: {
+    ...sharedOrderSelectProps,
+  },
+};
```

Comment:
Would this be the same as:

```suggestion
LoadingSearchResultListComponent.args = {
  ...BasicSearchResultListComponent.args,
  showSpinner: true,
};
```

Mind you, I also wonder if in `BasicSearchResultListComponent.args` you could also mostly spread from `EmptyResultListComponent.args`. Thoughts?

Reply:
Yeah, i think that makes sense. I will give it a go